scripts/outer/kubectl-kubestellar-prep_for_syncer

```diff
@@ -111,89 +111,67 @@ fi
 
-kubectl ws "${kubectl_flags[@]}" "${rootws}:${espw}"
-if ! cluster_ns_name="$(kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$spaceID" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
-	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$spaceID" >&2
+if ! cluster_ns_name="$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$imw" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
+	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$imw" >&2
 	exit 1
 fi
+
 prefixed_stname="$cluster_ns_name"-"$stname"
-stUID=$(kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
-mbwsname="$spaceID"-mb-"$stUID"
+stUID=$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
+mbwsname="$imw"-mb-"$stUID"
 
-espw_space_config="${PWD}/temp-space-config/spaceprovider-default-${espw}"
-kubectl-kubestellar-get-config-for-space --space-name ${espw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --space-config-file $espw_space_config
 if ! KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get APIExport edge.kubestellar.io &> /dev/null ; then
     echo "$0: it looks like ${espw@Q} is not the edge service provider workspace" >&2
     exit 2
 fi
 
-# TODO: mailbox does not support spaces yet
-if [ "$silent" == "true" ]; then
-    kubectl ws "${kubectl_flags[@]}" "$rootws" > /dev/null
-else
-    kubectl ws "${kubectl_flags[@]}" "$rootws"
-fi
-
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it a chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
 	echo "maybe the mailbox controller is slow, give it another chance"
     sleep 15
 fi
 
-if ! kubectl "${kubectl_flags[@]}" get Workspace "$mbwsname" &> /dev/null; then
+if ! KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" &> /dev/null; then
     echo "$0: did not find the mailbox workspace $mbwsname; is the mailbox controller running?" >&2
     exit 5
 fi
 
 echo "now that workspace _exists_, but is it _ready_?"
 sleep 5
 
-if ! kubectl ws "${kubectl_flags[@]}" "$mbwsname" &> /dev/null; then
+TESTMB_CMD="KUBECONFIG=$SM_CONFIG kubectl get space -n spaceprovider-default "$mbwsname" -o=jsonpath='{.status.Phase}'"
+
+
+if  phase=$(TESTMB_CMD); [ "$phase" != "READY" ]; then
```

Comment:
I doubt that this is the syntax you want.  Witness this from my Mac:

```
bash-3.2$ TESTMB_CMD="date"

bash-3.2$ phase=$(TESTMB_CMD)
bash: TESTMB_CMD: command not found
```

Witness this too:

```
bash-3.2$ phase=$($TESTMB_CMD)

bash-3.2$ echo $phase
Tue Dec 5 23:34:59 EST 2023

bash-3.2$ echo $($TESTMB_CMD)
Tue Dec 5 23:35:17 EST 2023
```

So I think you want something more like the following.

```
if  [ "$($TESTMB_CMD)" != READY ]; then
```

Reply:
Surprisingly there is also this:

```
bash-3.2$ TESTMB_CMD="echo 'foo   bar'"

bash-3.2$ echo $($TESTMB_CMD)
'foo bar'
```

so you might have some trouble with quoting. You might try defining a function rather than a variable that expands to a command.

Reply:
Better yet, just use `kubectl wait` as is done elsewhere.